pkg/middleware/instrumentation.go

```diff
@@ -0,0 +1,109 @@
+package middleware
+
+import (
+	"net/http"
+
+	"github.com/justinas/alice"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+var metricsHandler = NewInstrumentMetricHandlerWithDefaultRegistry()
```

Comment:
Instead, what about this?
```suggestion
var ServeMetrics = NewInstrumentMetricHandlerWithDefaultRegistry().ServeHTTP
```
Then in main where you have `middleware.MetricsServe` you replace it with `middleware.ServeMetrics`? (I prefer the name this way round)

Then we don't need the function wrapper at the end of this file

Reply:
Yeah, that makes more sense.

Reply:
I've stuck with MetricsHandler for now due to the change in handling shutdown in `main.go`. I'm happy to change if you think it should still be named differently.